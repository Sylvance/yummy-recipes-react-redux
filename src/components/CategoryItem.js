import React, { Component } from 'react'
import { ListGroup, ListGroupItem, ListGroupItemHeading, ListGroupItemText, Badge } from 'reactstrap';
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import CategoryEditFormInput from './CategoryEditFormInput'

export default class CategoryItem extends Component {
  static propTypes = {
    userid: PropTypes.number.isRequired,
    category: PropTypes.object.isRequired,
    editCategory: PropTypes.func.isRequired,
    deleteCategory: PropTypes.func.isRequired,
    fetchCategories: PropTypes.func.isRequired
  }

  state = {
    editing: false,
    user_id: this.props.userid
  }

  handleDoubleClick = () => {
    this.setState({ editing: true })
  }

  handleSave = (id, user_id, title, description) => {
    if (title.length === 0) {
      this.props.deleteCategory(id, user_id)
      this.props.fetchCategories(user_id)
    } else {
      this.props.editCategory(id, user_id, title, description)
      this.props.fetchCategories(user_id)
    }
    this.setState({ editing: false })
  }

  render() {
    const { category, deleteCategory } = this.props

    let element
    if (this.state.editing) {
      element = (
        <CategoryEditFormInput onSave={this.handleSave}
                          id={category.id}
                          user_id={this.state.user_id}
                          title={category.title}
                          description={category.description} />
      )
    } else {
      element = (
        <ListGroup>
          <ListGroupItem>
            <ListGroupItemHeading onDoubleClick={this.handleDoubleClick}>
              {category.title}
            </ListGroupItemHeading>
            <ListGroupItemText>
              {category.description}
              <br/>
              <a href={`/categories/${category.name}/${category.id}/recipes`}>View Recipes</a>
            </ListGroupItemText>
            <Badge pill color='danger' onClick={() => deleteCategory(category.id, this.state.user_id)}>
              Delete
            </Badge>
          </ListGroupItem>
        </ListGroup>
      )
    }

    return (
      <li className={classnames({
        editing: this.state.editing,
        item: true
      })}>
        {element}
      </li>
    )
  }
}
