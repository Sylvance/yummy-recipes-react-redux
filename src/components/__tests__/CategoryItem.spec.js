// Auto-generated do not edit

/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import CategoryItem from '../CategoryItem';


describe('CategoryItem test', () => {
  it('CategoryItem should match snapshot', () => {
    const component = renderer.create(
      <CategoryItem
        key={1} userid={1} category={{}} actions={{}}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
