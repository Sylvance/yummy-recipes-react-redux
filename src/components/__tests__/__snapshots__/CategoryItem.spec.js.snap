// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem test CategoryItem should match snapshot 1`] = `
<li
  className="item"
>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
    >
      <h5
        className="list-group-item-heading"
        onDoubleClick={[Function]}
      />
      <p
        className="list-group-item-text"
      >
        <br />
        <a
          href="/categories/undefined/undefined/recipes"
        >
          View Recipes
        </a>
      </p>
      <span
        className="badge badge-danger badge-pill"
        onClick={[Function]}
      >
        Delete
      </span>
    </li>
  </ul>
</li>
`;
